tests: add cmd/analytics integration test

eirinikos opened the pull request:
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

This change adds test coverage for `brew analytics`. 🎈

MikeMcQuaid commented:
Nice work 👍. Will let @UniqMartin do his usual 😉

UniqMartin commented on Library/Homebrew/test/test_integration_cmds.rb:
```diff
@@ -706,4 +706,23 @@ def test_irb
+      shutup do
+        system "git", "init"
+      end
+    end
```
Mostly out of curiosity: Why is the Git repository initialized in the middle of the test and how does (or doesn't) this influence the commands that are issued before and after this block?

eirinikos replied:
Hmmm. I had thought that the Git repository needed to be initialized before running `cmd("analytics", "regenerate-uuid")`, but this `cmd` seems to run smoothly regardless. I would have expected an error and I thought I remembered receiving one last week...

It does need to be initialized before toggling between `brew analytics on` and `brew analytics off`; running those `cmd`s without first initializing the repo does result in this error:

```
error: could not lock config file /tmp/homebrew-tests-20160726-97875-1wboued/prefix/.git/config:
No such file or directory
Error: Failure while executing:
git config --file=/tmp/homebrew-tests-20160726-97875-1wboued/prefix/.git/config
--replace-all homebrew.analyticsdisabled false[true]
```

In any case, I guess there's no solid reason for the Git repository to be initialized in the middle of the test. It seems to be fine to initialize it at the beginning. 😄

UniqMartin replied:
Thanks for the detailed response! I'm happy with the conclusion you drew from this (moving the repository initialization to the top of the test). :+1:

UniqMartin commented:
Didn't find anything to complain about (I know, I'm getting a bit out of my character :joy:), but I left a question as a code comment nonetheless.

:+1: Looks good to me!

eirinikos commented:
Thank you! I went ahead and moved the Git repository initialization to the beginning of the test (since there wasn't a compelling reason to keep it in the middle). And I rebased my branch to resolve the previously existing merge conflicts.

UniqMartin commented:
Good work! Thanks again @eirinikos! :tada:

MikeMcQuaid commented:
Thanks @eirinikos!
